src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -430,41 +439,96 @@ func (c controller) sendCompleteInstallation(isSuccess bool, errorInfo string) {
 	c.log.Infof("Done complete installation step")
 }
 
-// get pod logs,
-// write tar.gz to pipe in a routine
-// upload tar.gz from pipe to assisted service.
-// close read and write pipes
-func (c controller) uploadPodLogs(podName string, namespace string, sinceSeconds int64) error {
+func (c controller) uploadSummaryLogs(podName string, namespace string, sinceSeconds int64, isMustGatherEnabled bool) error {
+	var errz = make([]error, 0)
+	var tarentries = make([]utils.TarEntry, 0)
+	ctx := utils.GenerateRequestContext()
+
+	var returnVal = func() error {
```

Comment:
why we need this logic? Why just not to add log line to every error? I think it is too much for just printing log line

Reply:
its not for log lines at all. I want to be resilient for errors and collect the most logs I can. so if collection of must-gather fails I want to still be able to collect controller logs and vice versa. This is why I accumulate the errors instead of aborting the entire process at the first error we encounter - and then flatten them to a single error at the end.

Reply:
I think it is overcomplicated. Why u need to accumulate errors here? U do nothing with them.

Reply:
and u only return one at the end. The only thing u do is printing log in line 451 and nothing else